plugins/builddep.py

```diff
@@ -135,21 +135,29 @@ def _rpm_dep2reldep_str(rpm_dep):
         return rpm_dep.DNEVR()[2:]
 
     def _install(self, reldep_str):
-        sltr = dnf.selector.Selector(self.base.sack)
+        sltr_provides = dnf.selector.Selector(self.base.sack)
+        sltr_provides.set(provides=reldep_str)
+        sltrs = [sltr_provides]
+
         if reldep_str.startswith("/"):
-            sltr.set(file=reldep_str)
+            # Try also by file
+            sltr_files = dnf.selector.Selector(self.base.sack)
+            sltr_files.set(file=reldep_str)
+            sltrs.append(sltr_files)
+
+        for sltr in sltrs:
+            if not sltr.matches():
+                continue
+            already_inst = self.base._sltr_matches_installed(sltr)
+            if already_inst:
+                for package in already_inst:
+                    dnf.base._msg_installed(package)
+            self.base._goal.install(select=sltr, optional=False)
```

Comment:
Ok, it will work, but the approach is quit difficult to understand. I think you have to change work-flow, where you will test if selector is valid at the time of creation, or add proper description. The fact that search for validity of selector is done here ('if not sltr.matches():') and this is place where cpu is used is hard to understand. 
Also I have doubt about else: on line 157